backend/test/integration/langchain.test.ts

```diff
@@ -95,18 +99,16 @@ jest.mock('langchain/chains', () => {
 			return mockRetrievalQAChain;
 		}),
 		LLMChain: jest.fn().mockImplementation(() => {
-			return {
-				call: mockCall,
-			};
+			return mockPromptEvalChain;
 		}),
 	};
 });
-RetrievalQAChain.fromLLM = mockFromLLM;
+RetrievalQAChain.fromLLM =
+	mockFromLLM as unknown as typeof RetrievalQAChain.fromLLM;
```

Comment:
I suppose we can't avoid the `as unknown` ? I vaguely remember some discussions about it but can't remember where we landed.

Reply:
I think a cleverer person could avoid it.

As it stands though
- We want to mock this static method `RetrievalQAChain.fromLLM`
- `RetrievalQAChain.fromLLM` has its set type defined by the library (a function type which takes a bunch of complicated inputs and outputs an instance of `RetrievalQAChain`)
- Our mock will be a function that takes no inputs and outputs a particular mocked version of `RetrievalQAChain`. I want to keep it as simple as possible, and therefore it has a different type to 
- The one way that I've found to mock a static method so far that works is to import the object `RetrievalQAChain` and reassign its value explicitly `RetrievalQAChain.fromLLM = ...` [pretty sure there are other ways to do this]. But because the types differ, this requires casting, and because of some typescript or eslint quirk, it requires `as unknown as`

Reply:
Good to know, thanks!